### ISSUE TITLE
group_by modifier not working in v.3.3

### ISSUE THREAD

**Issue by sebszocinski** [statamic/cms#5761]
### Bug description

Using a grid field i'm trying to group by a subfield using the `group_by` modifier. This was working fine until upgrading to v3.3.

I understand the docs mention to use an alias (ie `{{ grid_field as="items" }}` and then use the modifier on the items loop ie  `{{ items group_by="somefield" }}`. This doesn't help.

The error we are getting is:
```
Facade\Ignition\Exceptions\ViewException
Method Statamic\Fields\Values::toAugmentedArray does not exist.
```

### How to reproduce

Using this data in your entry:

```
artists:
  -
     year: 2016
     artist: afed5139-6c57-46ed-b31f-359c08ebe25d
  -
     year: 2017
     artist: f70414dc-476c-40ee-99a8-b417205c2301
  -
      year: 2017
     artist: 1ac9c193-871e-4658-b32c-0136e3743d8f
  -
     year: 2018
     artist: e335857b-dc80-497a-947c-134142c09035
```

and this in your template/view:

```
{{ artists as="entries" }}
  {{ entries group_by="year" }}
    {{ groups }}
      {{ group }} - {{ artist }}{{ title }}{{ /artist }}
    {{ /groups }}
  {{ /entries }}
{{ /artists }}
```

You should get this error:

```
Facade\Ignition\Exceptions\ViewException
Method Statamic\Fields\Values::toAugmentedArray does not exist.
```

### Versions

Statamic v3.3.4
PHP 8
Laravel 8

### Installation

Existing Laravel app

### Antlers Parser

regex (default)

**Comment by sebszocinski** [comment]
I needed to patch this urgently for a client site so I dove into the core code and fixed it by [changing this line](https://github.com/statamic/cms/blob/a3c3e6ef5f13ce4a2d4ae9b7f4215afd32d85965/src/Modifiers/CoreModifiers.php#L819) from:
```
 $context = $item->toAugmentedArray($keys);
```
to:
```
$context = $item->toArray($keys);
```
It turns out that `$item` is returning a Laravel collection. I'm assuming it's expecting a Statamic Entry Collection? but this means it would only work on a Statamic relationship field and not a basic array ie a Grid field.

I'm totally guessing here, but hopefully this can be patched asap.

**Comment by jasonvarga** [comment]
I assume the template you've provided is a simplified version because it looks like you've also missed the `{{ items }} ... {{ /items }}` tag pair in there.

I could still replicate the error though and will be fixed by #5858.
